eli5/sklearn/transform.py

```diff
@@ -0,0 +1,179 @@
+                                        for n, w in zip(names, weights)
+                                        if w > threshold)
+                         for names, weights
+                         in zip(top_names.tolist(), top_weights.tolist()))
+        else:
+            all_feats = itertools.repeat('', weights.shape[0])
+
+        return [fmt.format(func_name=func_name, i=i, feats=feats)
+                for i, feats in enumerate(all_feats)]
+
+    return transform_names
+
+
+def transform_feature_names_polynomial_features(est, in_names=None):
+    # Thanks to @amueller in scikit-learn#6732
+    powers = est.powers_
+    if in_names is None:
+        in_names = _get_feature_names(est, feature_names=in_names,
+                                      num_features=powers.shape[1])
+    feature_names = []
+    for row in powers:
+        inds = np.where(row)[0]
+        if len(inds):
+            name = "*".join("%s^%d" % (in_names[ind], exp)
+                            if exp != 1 else in_names[ind]
+                            for ind, exp in zip(inds, row[inds]))
+        else:
+            name = "1"
+        feature_names.append(name)
+    return feature_names
+
+
+def transform_feature_names_binarizer(est, in_names=None):
+    # XXX: does not handle in_names=None case
+    fmt = '{{}}>{:.2g}'.format(est.threshold)
+    return [fmt.format(name) for name in in_names]
+
+
+def register_experimental_feature_names():
+    from sklearn.feature_extraction.text import TfidfTransformer
+    from sklearn.decomposition import (LatentDirichletAllocation,
+                                       TruncatedSVD,
+                                       PCA)
+    from sklearn.preprocessing import (Imputer,
+                                       StandardScaler,
+                                       RobustScaler,
+                                       MinMaxScaler,
+                                       Normalizer,
+                                       Binarizer,
+                                       OneHotEncoder,
+                                       PolynomialFeatures)
+
+    # By default these are ignored as all features are treated identically
+    @transform_feature_names.register(Imputer)
+    @transform_feature_names.register(StandardScaler)
+    @transform_feature_names.register(RobustScaler)
+    @transform_feature_names.register(MinMaxScaler)
+    @transform_feature_names.register(Normalizer)
+    def identity(est, in_names=None):
+        return in_names
+
+    transform_feature_names.register(Binarizer)(
+        transform_feature_names_binarizer)  # Perhaps should also be identity
+    transform_feature_names.register(PolynomialFeatures)(
+        transform_feature_names_polynomial_features)
+
+    # TODO: OneHotEncoder. scikit-learn#6441 doesn't appear complete
+
+    transform_feature_names.register(TfidfTransformer)(
+        make_tfn_featurewise('tfidf', lambda est: len(est.idf_)))
```

Comment:
This + https://github.com/TeamHG-Memex/eli5/commit/724c4232ba753e4ada84f16a0755d74a3e9a5750 should make mypy env pass. I haven't figured out how to make a pull request to your pull request :)